rero_ils/modules/stats_cfg/extensions.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2023 RERO
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Statistics configuration record extensions."""
+
+from invenio_records.extensions import RecordExtension
+from jsonschema.exceptions import ValidationError
+
+from rero_ils.modules.patrons.api import Patron, current_librarian
+from rero_ils.modules.utils import get_ref_for_pid
+
+
+class StatConfigDataExtension(RecordExtension):
+    """Add related statistics configuration data extension."""
+
+    categories = {
+        'number_of_documents': 'Catalogue',
+        'number_of_serial_holdings': 'Catalogue',
+        'number_of_items': 'Catalogue',
+        'number_of_patrons': 'User management',
+        'number_of_active_patrons': 'User management',
+        'number_of_ill_requests': 'Circulation',
+        'number_of_deleted_items': 'Catalogue',
+        'number_of_checkouts': 'Circulation',
+        'number_of_checkins': 'Circulation',
+        'number_of_renewals': 'Circulation',
+        'number_of_requests': 'Circulation'
+    }
+
+    def pre_commit(self, record):
```

Comment:
Do we really want to do this check? Why not, the new statistics will be different, ok but is it really a problem?

Reply:
If reports are created from a configuration and later on the configuration parameters are changed, it will no longer be possible to know the values of the configuration parameters used to create these reports. 
I would leave a check on the main fields ('indicator', 'dist1', 'dist2', 'filters') but I can remove 'organisation' and 'category' because these fields are created by the pre_create and not set by the user. What do you think ?

Reply:
@zannkukai I agree with you, this is a POs decision. 
I think that once a report is created we should prevent from editing any parameter of the configuration that changes the results of the query, this includes 'indicator', 'dist1', 'dist2', 'period', 'filter' (and arguably also 'frequency'). We should allow to change only 'name', 'description' and 'status'. 
If you agree I will remove this constrain for now and we will add it to another PR once we have an answer from the POs.